app/controllers/providers/edit_initial_allocations_controller.rb

```diff
@@ -0,0 +1,119 @@
+module Providers
+  class EditInitialAllocationsController < ApplicationController
+    def do_you_want
```

Comment:
Interesting method name 😝

Reply:
I agree with @TobyRet that this is ... "interesting". :) Controllers are usually named after resources and methods are usually actions, so I'd expect that these are in the existing `AllocationsController` with a service object to wrap up all the messiness, or if really need be, in a `InitialAllocationController` or something that expresses the thing being worked on, instead of the actions being performed.